.github/workflows/release.yml

```diff
@@ -0,0 +1,180 @@
+        required: false
+        default: "y"
+      skipGitHub:
+        description: Should we skip publishing artefacts to GitHub Packages
+        required: false
+        default: "y"
+  release:
+    types: [ created ]
+
+env:
+  GRADLE_OPTS: "-Dorg.gradle.daemon=true"
+
+jobs:
+  check:
+    uses: ./.github/workflows/check.yml
+
+  resolve-version:
+    uses: ./.github/workflows/resolve-version.yml
+    needs: [ check ]
+    with:
+      desired-version: ${{ github.event.release.tag_name || github.event.inputs.version }}
+
+  build-dokka:
+    name: Build Dokka
+    needs:
+      - check
+      - resolve-version
+    runs-on: ubuntu-latest
+    env:
+      VERSION: ${{ needs.resolve-version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 11
+
+      - name: Restore Gradle cache
+        id: cache-gradle
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.gradle/yarn
+            ~/.gradle/nodejs
+            ~/.konan
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Gradle Assemble Dokka
+        run: ./gradlew dokkaHtml -Pversion=${VERSION//v} --scan
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: dokka
+          path: |
+            **/build/dokka
+
+  release-Artefacts:
+    name: Release to ${{ matrix.repository.name }} on ${{ matrix.os.runner }}
+    runs-on: ${{ matrix.os.runner }}
+    needs:
+      - check
+      - resolve-version
+    defaults:
+      run:
+        shell: ${{ matrix.os.shell }}
+    env:
+      VERSION: ${{ needs.resolve-version.outputs.version }}
+      GH_USERNAME: ${{ github.actor }}
+      GH_PASSWORD: ${{ github.token }}
+      ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_SECRET }}
+      ORG_GRADLE_PROJECT_signingKeyId: ${{ secrets.GPG_SIGNING_KEY_ID }}
+      ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.GPG_SIGNING_PASSWORD }}
+      ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+      ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+    strategy:
+      matrix:
+        repository:
+          - name: Github Packages
+            tasks: publishAllPublicationsToGitHubRepository
+            enabled: ${{ github.event.inputs.skipGitHub != 'y' }}
+          - name: Maven Central
+            # `closeSonatypeStagingRepository` Requires to manually release a new version on https://s01.oss.sonatype.org/#stagingRepositories
+            # If you're brave, you could change it to `closeAndReleaseSonatypeStagingRepository` to fully automate the release
+            tasks: publishToSonatype closeAndReleaseSonatypeStagingRepository
```

Comment:
If this can fully automate release publishing I'm ok with that for now.  Potentially risk publishing busted artifacts, but given we are all busy I think it's worth the trade off for now.

Reply:
I've added `publishToLoacal` task to our check runs to at least be sure we can build all the artefacts fine. Let's keep it until the first error and discuss alternatives if that happens.